### DISCUSSION TITLE
Get Task/Event from Child Object not in Master-Detail relationship

### DISCUSSION THREAD

**Discussion by AnseGit** [deejay-hub/timeline-lwc#295] [Q&A] [answered]
Hi 
We have the need to fetch Task/Events from a child record not related through a Master Detail relationship. 
Is there any Out-of-the--box way of doing this? 

Example: 
We have a custom object (customObject__c) that is related to Account/Person Account through a standard Lookup-field (Not master detail).
As this is not master-detail relationsship, we are not seeing the task/Events in the Activity timeline at the account.  But I'd like to add activities added on CustomObject__c to the Timewarp of the Account.

Here is the Soql i need to be able to retrieve: 

SELECT Id, Subject, WhatId
FROM Task
WHERE WhatId IN (SELECT Id FROM CustomObject__c WHERE Account__c = 'AccountId')

How can this be done?

**Comment by deejay-hub** [comment]
The fact that it is Master Detail or Lookup isn't important - Time Warp works for both types. The following Timeline_Configuration__mdt type record would plot what you want

<img width="1323" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/69f760b1-7656-41bb-a78d-6eaf6ebcbb6f">

This will plot the Task like this

<img width="1700" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/a1a2a4d3-c0dc-43e6-b797-7dc0daa3de28">

With the Task like this - showing the RelatedTo field populated with the parent CustomObject__c record

<img width="1692" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/439df5ce-311c-4423-8618-17a078c6f5c4">

Make sure the Task
- Has a date value populated based on the field you use in the Position Date field in the metadata

Make sure the Metadata record
 - Is Active
 - Is populated with the name of the custom object correctly

**Comment by AnseGit** [comment]
Hi 
The solution you are suggesting works for the timeline on the Custom Object, but my timeline is placed on the Account that the custom Object is related to through a standard lookup.field. 

So in the example you have presented above, how can this task (Test Task) that is related to CustomObject__c, be displayed in the timeline at the Account that is related to this custom object through the standard lookupfield field CustomObject__r.Account__c?

**Comment by deejay-hub** [comment, marked as answer]
Time Warp doesn't support rollups to the Account level for custom objects. You would need to customize the code yourself for that use case. Or add a field to the parent object on the Activity object and populate that field.

**Reply by AnseGit** [reply]
I understand! Thank you for the help :)
